crates/libs/windows/src/core/agile_reference.rs

```diff
@@ -0,0 +1,30 @@
+use super::*;
+use bindings::*;
+use core::marker::PhantomData;
+
+/// A safe wrapper around WinRT interfaces.
+///
+/// Some interfaces are not marked as agile thus then don't implement [`Send`].
+/// These interfaces can be made agile through this `AgileReference`.
+pub struct AgileReference<T> {
+    reference: IAgileReference,
+    _marker: PhantomData<T>,
+}
```

Comment:
```suggestion
#[repr(transparent)]
#[derive(Clone, PartialEq, Eq)]
pub struct AgileReference<T> (IAgileReference, PhantomData<T>);
```

For consistency with the rest of `windows::core`, let's define it as such.